lib/matplotlib/legend_handler.py

```diff
@@ -603,31 +602,44 @@ def create_artists(self, legend, orig_handle,
         else:
             bottom = self._bottom
 
-        leg_markerline = Line2D(xdata_marker, ydata[:len(xdata_marker)])
-        self.update_prop(leg_markerline, markerline, legend)
-
         leg_stemlines = []
-        for thisx, thisy in zip(xdata_marker, ydata):
-            l = Line2D([thisx, thisx], [bottom, thisy])
-            leg_stemlines.append(l)
 
-        for lm, m in zip(leg_stemlines, stemlines):
-            self.update_prop(lm, m, legend)
+        # update_prop() usually takes two Line2D collections;
+        # override temporarily to copy properties from a LineCollection
+        orig_update_func = self._update_prop_func
+        self._update_prop_func = self._copy_collection_props
+
+        for thisx, thisy in zip(xdata_marker, ydata):
+            thisline = Line2D([thisx, thisx], [bottom, thisy])
+            leg_stemlines.append(thisline)
+            self.update_prop(thisline, stemlines, legend)
+        # Reset update_prop_func
+        self._update_prop_func = orig_update_func
 
         leg_baseline = Line2D([np.min(xdata), np.max(xdata)],
                               [bottom, bottom])
 
         self.update_prop(leg_baseline, baseline, legend)
 
-        artists = [leg_markerline]
-        artists.extend(leg_stemlines)
+        leg_markerline = Line2D(xdata_marker, ydata[:len(xdata_marker)])
+        self.update_prop(leg_markerline, markerline, legend)
+
+        artists = leg_stemlines
```

Comment:
I think @anntzer would say that you can use Python 3 syntax now:
`artist = [*leg_stemlines, leg_baseline, leg_markerline]`